src/App.tsx

```diff
@@ -15,36 +15,102 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+type SortField = 'alphabet' | 'length' | '';
+
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState<string[]>(goodsFromServer);
+  const [isActive, setIsActive] = useState<SortField>();
+  const [reverse, setReverse] = useState(false);
+
+  const resetSort = () => {
+    setGoods([...goodsFromServer]);
+    setIsActive('');
+    setReverse(false);
+  };
+
+  const reversGoods = (foods: string[]) => {
+    setReverse(!reverse);
+    setGoods([...foods].reverse());
+  };
+
+  const sortedGoods = (type: SortField): void => {
+    setGoods(
+      [...goods].sort((good1, good2) => {
+        if (type === 'alphabet') {
+          setIsActive(type);
+          if (reverse) {
+            return good2.localeCompare(good1);
+          }
+
+          return good1.localeCompare(good2);
+        }
+
+        if (type === 'length') {
+          setIsActive(type);
+          if (reverse) {
+            return good2.length - good1.length;
+          }
+
+          return good1.length - good2.length;
+        }
+
+        return 0;
+      }),
+    );
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          onClick={() => {
+            sortedGoods('alphabet');
+          }}
+          type="button"
+          className={`button is-info ${isActive !== 'alphabet' ? 'is-light' : ''} `}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          onClick={() => {
+            sortedGoods('length');
+          }}
```

Comment:
Create a enum for 'length'
```suggestion
          onClick={() => sortedGoods('length')}
```

Reply:
https://github.com/mate-academy/react_list-of-goods-hooks/pull/1913#discussion_r1640788422